trustfall_core/src/schema/adapter/mod.rs

```diff
@@ -73,6 +73,61 @@ impl<'a> SchemaAdapter<'a> {
     pub fn schema_text() -> &'static str {
         include_str!("./schema.graphql")
     }
+
+    pub fn vertex_type_iter(
```

Comment:
This is definitely not a fit to be a `pub fn` (we don't want to commit to publicly maintaining it forever) and probably also shouldn't be a method on the adapter either. It can just as easily be an independent private function that takes a `&Schema` argument.